packages/cosmos/src/client.ts

```diff
@@ -284,6 +284,20 @@ export class CosmosClient {
     this.setHeaders(request.headers, headers);
     request.headers.set('content-type', 'application/query+json');
     const response = await this.fetchWithRetry(request);
+    if (
+      (await response.clone().text()).includes(
+        'The provided cross partition query can not be directly served by the gateway'
```

Comment:
Were there any response headers that would indicate this status? It would be great if we did not need to clone and read the response text. Sometimes there's an `x-ms-substatus` header for example.

Reply:
As far as I know no, I will double check to be sure.

Reply:
Nope no headers, sucks...

Reply:
Here is an example response:

```json
{
    "raw": {
        "webSocket": null,
        "url": "https://REDACTED-cosmos.documents.azure.com/dbs/REDACTED/colls/listings/docs",
        "redirected": false,
        "ok": false,
        "headers": {},
        "statusText": "Bad Request",
        "status": 400,
        "bodyUsed": true,
        "body": {
            "locked": true
        }
    },
    "headers": {},
    "text": "{\"code\":\"BadRequest\",\"message\":\"The provided cross partition query can not be directly served by the gateway. This is a first chance (internal) exception that all newer clients will know how to handle gracefully. This exception is traced, but unless you see it bubble up as an exception (which only happens on older SDK clients), then you can safely ignore this message.\\r\\nActivityId: 4bf6aa28-e93b-4841-955f-f7af7672166b, Microsoft.Azure.Documents.Common/2.14.0\",\"additionalErrorInfo\":\"{\\\"partitionedQueryExecutionInfoVersion\\\":2,\\\"queryInfo\\\":{\\\"distinctType\\\":\\\"None\\\",\\\"top\\\":6,\\\"offset\\\":null,\\\"limit\\\":null,\\\"orderBy\\\":[\\\"Descending\\\"],\\\"orderByExpressions\\\":[\\\"c.price\\\"],\\\"groupByExpressions\\\":[],\\\"groupByAliases\\\":[],\\\"aggregates\\\":[],\\\"groupByAliasToAggregateType\\\":{},\\\"rewrittenQuery\\\":\\\"SELECT TOP 6 c._rid, [{\\\\\\\"item\\\\\\\": c.price}] AS orderByItems, c AS payload\\\\nFROM c\\\\nWHERE ({documentdb-formattableorderbyquery-filter})\\\\nORDER BY c.price DESC\\\",\\\"hasSelectValue\\\":false,\\\"dCountInfo\\\":null},\\\"queryRanges\\\":[{\\\"min\\\":\\\"\\\",\\\"max\\\":\\\"FF\\\",\\\"isMinInclusive\\\":true,\\\"isMaxInclusive\\\":false}]}\"}"
}
```

Reply:
I think there should be an `x-ms-substatus: 1004` header. https://github.com/Azure/azure-cosmosdb-node/blob/ef4b113757b72356475f6bad994dc87f17e49291/source/lib/statusCodes.js#L64

Reply:
I am reading headers from response.raw.headers and as you can see there are none, so idk maybe this sdk is messing with the headers?

Reply:
Nevermind I'm stupid and was assuming that stringifying the headers object would give them. Parsing them gives this:
```json
{
	"1004": "x-ms-substatus",
	"DYNAMIC": "cf-cache-status",
	"6b482097c1b0fb80-AKL": "cf-ray",
	"keep-alive": "connection",
	"application/json": "content-type",
	"Sat, 27 Nov 2021 03:04:07 GMT": "date",
	"max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\"": "expect-ct",
	"cloudflare": "server",
	"max-age=31536000": "strict-transport-security",
	"chunked": "transfer-encoding",
	"77434ca6-4167-483a-b32b-d3370dd6a93c": "x-ms-activity-id",
	"version=2.14.0": "x-ms-gatewayversion"
}
```

Reply:
Done, see latest commit